pytket/extensions/qujax/qujax_convert.py

```diff
@@ -123,6 +145,145 @@ def tk_to_qujax_symbolic(
             jnp.array([symbol_map[symbol] for symbol in c.op.free_symbols()])  # type: ignore
         )
 
-    return get_params_to_statetensor_func(
-        gate_name_seq, qubit_inds_seq, param_inds_seq, circuit.n_qubits
+    return gate_name_seq, qubit_inds_seq, param_inds_seq, circuit.n_qubits
+
+
+def tk_to_qujax_symbolic(
+    circuit: Circuit, symbol_map: Optional[dict] = None
+) -> qujax.UnionCallableOptionalArray:
+    """
+    Converts a tket circuit with symbolics parameters and a symbolic parameter map
+    into a function that maps circuit parameters to a statetensor.
+    Assumes all circuit gates can be found in qujax.gates.
+    Note that the behaviour of tk_to_qujax_symbolic(circuit)
+    is different to tk_to_qujax(circuit),
+    tk_to_qujax_symbolic will look for parameters in circuit.free_symbols()
+    and if there are none it will assume that none of the gates require parameters.
+    On the other hand, tk_to_qujax will work out which gates are parameterised
+    based on e.g. circuit.get_commands()[0].op.params
+
+    :param circuit: Circuit to be converted.
+    :type circuit: pytket.Circuit
+    :param symbol_map: dict that maps elements of circuit.free_symbols() (sympy)
+        to parameter indices.
+    :type symbol_map: Optional[dict]
+    :return: Function which maps parameters
+        (and optional statetensor_in) to a statetensor.
+        If the circuit has no parameters, the resulting function
+        will only take the optional statetensor_in as an argument.
+    :rtype: Callable[[jnp.ndarray, jnp.ndarray = None], jnp.ndarray]
+        or Callable[[jnp.ndarray = None], jnp.ndarray]
+        if no parameters found in circuit
+    """
+    return qujax.get_params_to_statetensor_func(
+        *tk_to_qujax_args_symbolic(circuit, symbol_map)
     )
+
+
+def print_circuit(
```

Comment:
Why is this a function in pytket-qujax? What is the benefit from having this in pytket-qujax and not only in qujax?

Reply:
I thought it would be useful to be able to print out a string representation of the circuit (i.e. when working in console not jupyter)

qujax does not require pytket so the pytket.Circuit -> print circuit functionality has to be hosted on the extension

Reply:
I thought this might be better handled with a print of the converted circuit. So only offering the conversion in pytket-qujax and using the qujax print function? Not sure if this is a function that I would recommend to use, because the user can do something else with the circuit then printing the result of the conversion? The result of the conversion is hidden and if they want to do something else they would need to repeat the conversion?

Reply:
I think having both functionalities is ok, it is cumbersome to carry around the qujax_args and a key goal of this extension is to have the functionality of qujax indexed via a pytket.Circuit

My idea behind print_circuit is just a quick sanity check that the circuit is what you want it to be, on that basis I enjoy having the ability to do
`print_circuit(circuit)`
rather than
`qujax_args = tk_to_qujax_args(circuit)`
`print_circuit(*qujax_args)`

So really I see the main functions being `tk_to_qujax` and `print_circuit` with `tk_to_qujax_args` there for completeness and in case users do not want to repeat the conversion (although repeating the conversion isn't so bad as it's very fast)

Reply:
This question is what the sanity check is helping if you can't use what you have checked? If you want to do a check, there should not be the need to trust conversion function to give the same result every time?

Reply:
Especially for symbolic circuits there could be a change depended in the parameters given?

Reply:
the print_circuit function won't actually display the values of the parameters only the index of the input parameter array in which to find it for each gate

so the `print_circuit` is just a sanity check that the `pytket.Circuit` (and possibly `symbol_map`) have been constructed as intended. If something is amiss the `pytket.Circuit` can then be adjusted before being sent to `tk_to_qujax` which constructs the full JAX function

The user then doesn't ever have to touch any qujax_args (unless they want to)

Reply:
Okay, thank you for making that clear!